.travis.yml

```diff
@@ -11,18 +11,16 @@ jobs:
 
   - stage: docker build and publish
     script:
```

Comment:
Note that this step is still running every time; which avoids the main problem (deploying every commit) but could still cause issues for other users if they expect Dockerhub's image tagged `latest` to be a reliable one; I'm happy with that here but do consider how we might see issues, and how we might change accordingly;
* We could treat the image in DockerHub as a dev image, and not worry about it!
* We could push a tagged image (maybe with a commit hash?) every time, but only the `latest` when we're deploying
* We could switch to push the latest image only when actually deploying